### PR TITLE
Return X-Powered-By header #1224

### PR DESCRIPTION
**Pull request by matiaskusack**
This is my first PR, not sure if I'm doing it right. 

Implementation of this: https://github.com/OrchardCMS/OrchardCore/issues/1224.

Created a middleware to inject X-Powered-By header, extensions to ModularApplicationBuilder and test methods.

### CODE REVIEW

File: src/OrchardCore/OrchardCore.Modules/PoweredByMiddleware.cs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        /// <param name="httpContext">The Http Context</param>
+        /// <param name="headerName">Header name</param>
+        /// <param name="headerValue">Header value</param>
+        private static void AddHeader(HttpContext httpContext, string headerName, string headerValue)
+        {
+            httpContext.Response.OnStarting(() =>
+            {
+                var headers = httpContext.Response.Headers;
+
+                if (!headers.ContainsKey(headerName))
```
Comment (sebastienros) [inline src/OrchardCore/OrchardCore.Modules/PoweredByMiddleware.cs]: Can it be written ` headers[headerName] = headerValue;` without any key lookup?

**Comment by sebastienros** [comment]
I have simplified the code to add the header, and it will also be faster this way.
But I have broken your unit tests. Please fix them of just remove them, but they could be simpler if you didn't have to mock the headers dictionary. Thanks anyway, that's some serious work for such a little feature.

**Comment by sebastienros** [comment]
Also you are working your fork's dev branch. Next time you should create a feature branch first, this way you will be able to work on different PRs at the same time, or update the dev branch without altering existing PRs. It's not an issue right now but you got lucky.

File: src/OrchardCore/OrchardCore.Modules/PoweredByMiddleware.cs (hunk 1 above)
Comment (matiaskusack) [reply src/OrchardCore/OrchardCore.Modules/PoweredByMiddleware.cs]: Well, i thought that if there's another middleware inserting the header first, we should append our Powered By. Katana for example it's doing it adding a new header either way and it results on two Powered-By headers.
Anyway its just works, and yours its a KISS solution, so I agree on that.

**Comment by matiaskusack** [comment]
Ok, I'll fix the tests. Ok on the feature branch too. 
Thanks

**Pull request merged**